cpp/src/parquet/encoding_test.cc

```diff
@@ -862,5 +862,215 @@ TEST_F(DictEncoding, CheckDecodeIndicesNoNulls) {
   CheckDict(actual_num_values, *builder);
 }
 
+// ----------------------------------------------------------------------
+// BYTE_STREAM_SPLIT encode/decode tests.
+
+namespace {
+template <typename DType>
+void TestEncodeDecodeWithBigInput() {
+  const int nvalues = 10000;
+  using T = typename DType::c_type;
+  std::vector<T> data(nvalues);
+  GenerateData<T>(nvalues, data.data(), NULLPTR);
+  std::unique_ptr<TypedEncoder<DType>> encoder =
+      MakeTypedEncoder<DType>(Encoding::BYTE_STREAM_SPLIT);
+  encoder->Put(data.data(), static_cast<int>(data.size()));
+  std::shared_ptr<Buffer> buffer = encoder->FlushValues();
+
+  std::unique_ptr<TypedDecoder<DType>> decoder =
+      MakeTypedDecoder<DType>(Encoding::BYTE_STREAM_SPLIT);
+  decoder->SetData(static_cast<int>(data.size()), buffer->mutable_data(),
+                   static_cast<int>(buffer->size()));
+  std::vector<T> decoded_data(nvalues);
+  int num_decoded_elements = decoder->Decode(decoded_data.data(), nvalues);
+  ASSERT_EQ(nvalues, num_decoded_elements);
+  for (size_t i = 0U; i < decoded_data.size(); ++i) {
+    ASSERT_EQ(data[i], decoded_data[i]);
+  }
```

Comment:
Seems like there is a missing helper function for checking a data round trip given an encoder and decoder implementing the abstract interface. Do you want to try to refactor?

Reply:
Yes, I can write a separate patch.

Reply:
It would be nice to at least add the roundtrip helper(s) in this PR, since otherwise there's a lot of additional duplication.